runtime/notification.go

```diff
@@ -0,0 +1,71 @@
+package runtime
+
+import (
+	"fmt"
+
+	"github.com/mitchellh/mapstructure"
+)
+
+type Notification struct {
+	GroupID   string      `json:"group_id"`
+	EventType string      `json:"event_type"`
+	Event     interface{} `json:"event"`
+}
+
+type StageEvent struct {
+	Name   string `json:"name"`
+	Action string `json:"action"` // entry or exit
+}
```

Comment:
I would just simplify all of this and have a single `Event` struct that has a string-based type (`type EventType string`), with a bunch of enumerated presets that follow the `Record*` naming, as well as `stage:start`, `stage:end:ok`, `stage:end:err` events.

Reply:
Not sure if I understand exactly what you mean.

I don't like the flat model, as it is not explicit about the fields that are required/optional for a certain type of event.

`StageEvent` is really nothing like an `OutcomeEvent`, and placing then within the same fields (or having redundant fields) is suboptimal IMO.